src/treelearner/cuda/cuda_single_gpu_tree_learner.cpp

```diff
@@ -219,13 +257,17 @@ Tree* CUDASingleGPUTreeLearner::Train(const score_t* gradients,
     SelectFeatureByNode(tree.get());
 
     if (config_->use_quantized_grad) {
-      const uint8_t smaller_leaf_num_bits_bin = cuda_gradient_discretizer_->GetHistBitsInLeaf<false>(smaller_leaf_index_);
-      const uint8_t larger_leaf_num_bits_bin = larger_leaf_index_ < 0 ? 32 : cuda_gradient_discretizer_->GetHistBitsInLeaf<false>(larger_leaf_index_);
+      const uint8_t smaller_leaf_num_bits_bin = nccl_communicator_ == nullptr ?
+        cuda_gradient_discretizer_->GetHistBitsInLeaf<false>(smaller_leaf_index_) :
+        cuda_gradient_discretizer_->GetHistBitsInLeaf<true>(smaller_leaf_index_);
+      const uint8_t larger_leaf_num_bits_bin = larger_leaf_index_ < 0 ? 32 : (nccl_communicator_ == nullptr ?
```

Comment:
`32` is always used despite the value of `num_grad_quant_bins` parameter value, right?